pgctl/cli.py

```diff
@@ -275,6 +277,35 @@ def aliases(self):
             'default': self.all_services
         })
 
+    def ensure_scratch_dir_exists(self, service_path):
+        """Ensure that the scratch directory exists and symlinks supervise.
+
+        Due to quirks in pip and potentially other package managers, we don't
+        want named FIFOs on disk inside the project repo (they'll end up in
+        tarballs and other junk).
+
+        Instead, we stick them in a scratch directory outside of the repo.
+        """
+        home_path = self.scratch_dir(service_path)
+
+        # ensure symlink {service_dir}/supervise -> {scratch_dir}/supervise
+        supervise_in_scratch = home_path.join('supervise')
+        supervise_in_scratch.ensure_dir()
+
+        supervise_in_repo = service_path.join('supervise')
+        if supervise_in_repo.exists():
+            if supervise_in_repo.islink():
+                # The user probably moved the repo, relinking is pretty safe.
+                if supervise_in_repo.readlink() != supervise_in_scratch.strpath:
+                    supervise_in_repo.remove()
+            else:
+                raise ValueError('{} exists and is not a symlink, please remove it!'.format(
+                    supervise_in_repo.strpath,
+                ))
```

Comment:
I'd just run some equivalent of `ln -sf` and throw a stacktrace if someone was wonky.

Personally I'd literally run `ln -sf`. I dont plan windows support any day soon.